File: Struts/StudentCRUD/src/main/java/com/techlabs/service/StudentService.java
package com.techlabs.service;

import java.util.List;
import com.techlabs.dao.StudentDao;
import com.techlabs.model.Student;

public class StudentService {

	private StudentDao studentDao;

	public StudentService() {

		studentDao = StudentDao.getStudentService();
	}

	public Student getStudent(String rollNo) {

		return studentDao.getStudent(rollNo);
	}

	public List<Student> getStudents() {

		return studentDao.getStudents();
	}

	public void addStudent(String name, String address, int age) {

		studentDao.addStudent(name, address, age);
	}

	public void updateStudent(String rollNo, String name, String address, int age) {

		studentDao.updateStudent(rollNo, name, address, age);
	}

	public void deleteStudent(String rollNo) {

		studentDao.deleteStudent(rollNo);
	}

	public List<Student> searchStudents(String name) {

		return studentDao.searchStudents(name);
	}
}

File: Angular CLI/student-app-new/src/app/services/student.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Observable } from 'rxjs';
import { Student } from "../classes/student";

@Injectable({
  providedIn: 'root'
})
export class StudentService {

  students:Student[] = [];
  constructor(private httpClient:HttpClient){}
  baseUrl:string = "http://gsmktg.azurewebsites.net/api/v1/techlabs/test/students/";

  getStudents():Observable<Student[]>{
    return this.httpClient.get<Student[]>(this.baseUrl);
  }
  
  addStudent(student:Student):Observable<Student>{
    
    let studentJSON:string = JSON.stringify(student);
    let httpHeaders = new HttpHeaders({'Content-Type' : 'application/json'}); 
    console.log(studentJSON);

    return this.httpClient.post<Student>(this.baseUrl, studentJSON, {'headers':httpHeaders});
  }

  getStudent(id:string):Observable<Student>{
    return this.httpClient.get<Student>(this.baseUrl + id);
  }

  updateStudent(student:Student):Observable<Student>{
    let studentJSON:string = JSON.stringify(student);
    let httpHeaders = new HttpHeaders({'Content-Type' : 'application/json'}); 
    return this.httpClient.put<Student>(this.baseUrl + student.id, studentJSON, {'headers':httpHeaders});
  }
  deleteStudent(id:string){
    return this.httpClient.delete<Student>(this.baseUrl + id);
  }
}

